### PR TITLE
Adding Spire Federation

### PR DESCRIPTION
**Pull request by akanshasajimon**
Adding spire quickstart , spiffeids for federation and federation trust domain for multiple clusters.

### CODE REVIEW

File: services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  trustDomainBundle: |- #replace keys with trust bundle from cluster 2
+    {
+      "keys": [
+        {
+            "use": "x509-svid",
+            "kty": "EC",
+            "crv": "P-256",
+            "x": "7ZZmZoyTep5ssVTmZMw3sMQUjkOdxQTq2b0R5rTVlHM",
+            "y": "7BiKMky421_ggato-vff6Wv5UaJZm4CaGsp-t2DvFx4",
+            "x5c": [
```
Comment (sudhakso) [inline services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml]: Mask the keys. Replace it with the palceholder like <Get your keys from cluster using command...>

File: services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apiVersion: spire.spiffe.io/v1alpha1
+kind: ClusterFederatedTrustDomain
+metadata:
+  name: cluster2.demo
```
Comment (sudhakso) [inline services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml:4]: Mask this value also as per the name of your cluster. Document how to retrieve the cluster name

File: services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+spec:
+  trustDomain: cluster2.demo
+  bundleEndpointURL: https://172.16.17.3:8443 #replace with ip of spire server endpoint on cluster 2
```
Comment (sudhakso) [inline services/spire/federation/clusterfederatedtrustdomain-cluster1.yaml]: Again mask this value as bundleEndpointURL: <Replace with IP of sprire server, e.g. https://172.16.17.3:8443>

**Review by sudhakso — APPROVED** [review]

**Review by GuopingJia — APPROVED** [review]
LGTM

**Pull request merged**